lib/resty/upload.lua

```diff
@@ -169,20 +168,9 @@ end
 
 
 local function read_header(self)
-    local read_line = self.read_line
-
-    local line, err = read_line(MAX_LINE_SIZE)
-    if err then
-        return nil, nil, err
-    end
-
-    local dummy, err = read_line(1)
-    if dummy then
-        return nil, nil, "line too long: " .. line .. dummy .. "..."
-    end
-
-    if err then
-        return nil, nil, err
+    local line, err = discard_line(self)
```

Comment:
This looks logically wrong. We actually need that line and we are definitely not "discarding" it here at all.